tt_metal/impl/device/device.cpp

```diff
@@ -1863,7 +1870,14 @@ void Device::init_command_queue_host() {
 
 void Device::init_command_queue_device() {
 
-    this->compile_command_queue_programs();
+    if (llrt::OptionsG.get_skip_loading_fw()) {
+        detail::EnablePersistentKernelCache();
+        this->compile_command_queue_programs();
+        detail::DisablePersistentKernelCache();
```

Comment:
Should it always set disable after? What if user had it enabled?

Reply:
Should probably check if it wasn't enabled by the user (default behaviour I believe) before disabling it.

Reply:
That would be a bug. The persistent cache state isn't a variable I can query here, so I can't restore it to what it was previously.